### PR TITLE
Fix nested prefabs with custom MRTK inspectors

### PR DESCRIPTION
**Pull request by Troy-Ferrell**
## Overview
When creating custom controls via EditorGUILayout.blah that isn't a property field, unity is not able to correctly track changes between the original component in the prefab and modifications at the scene layer (or higher level prefab layers). Usually Unity out-of-box highlights/bolds the modified field and allow right click options to control the property. 

This change is to go through the custom inspector and identify where MRTK code needs to use EditorGUILayout.PropertyField or use the EditorGUI.PropertyScope to enable this functionality back in Unity.

## Changes
- Fixes: #4369

### CODE REVIEW

**Comment by Troy-Ferrell** [comment]
Friendly ping to reviewers @julenka @keveleigh  @davidkline-ms  @thalbern .
This PR has been out for 3 weeks.

File: Assets/MixedRealityToolkit.SDK/Inspectors/Input/Handlers/ControllerPoseSynchronizerInspector.cs [hunk 1]
```diff
@@ -58,17 +59,14 @@ public override void OnInspectorGUI()
...
                 {
-                    var currentHandedness = (Handedness)handedness.enumValueIndex;
-                    var handIndex = currentHandedness == Handedness.Right ? 1 : 0;
-
-                    using (var c = new EditorGUI.ChangeCheckScope())
+                    Rect position = EditorGUILayout.GetControlRect();
+                    var label = new GUIContent(handedness.displayName);
+                    using (new EditorGUI.PropertyScope(position, label, handedness))
                     {
-                        var newHandednessIndex = EditorGUILayout.Popup(handedness.displayName, handIndex, HandednessLabels);
```
Comment (keveleigh) [inline Assets/MixedRealityToolkit.SDK/Inspectors/Input/Handlers/ControllerPoseSynchronizerInspector.cs]: nit: looks like `HandednessLabels` isn't used anymore. Can it be removed?

**Pull request merged**